server/src/main/java/org/elasticsearch/cluster/routing/allocation/AllocateUnassignedDecision.java

```diff
@@ -267,6 +267,7 @@ public String getExplanation() {
         checkDecisionState();
         return switch (getAllocationDecision()) {
```

Comment:
Looking at `AllocateUnassignedDecision#getAllocationDecision` it appears as though it'll never return `NOT_PREFERRED`,  see `AllocationDecision#fromAllocationStatus`